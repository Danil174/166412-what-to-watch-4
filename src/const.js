export const TextMovieRatings = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
};

export const SmallCardVideoSettings = {
  WIDTH: `100%`,
  HEIGHT: `100%`,
  IS_MUTED: true,
};

export const smallCardVideoStartTimeOut = 1000;
