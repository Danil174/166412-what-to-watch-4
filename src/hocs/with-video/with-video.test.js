import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withVideo from "./with-video.js";
import {SmallCardVideoSettings} from "../../const.js";


const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withVideo(MockComponent);

it(`withVideo is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped
      preview={`https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jp`}
      source={`http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`}
      isMuted={SmallCardVideoSettings.IS_MUTED}
      width={SmallCardVideoSettings.WIDTH}
      height={SmallCardVideoSettings.HEIGHT}
      isPlaying={false}
    />
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
