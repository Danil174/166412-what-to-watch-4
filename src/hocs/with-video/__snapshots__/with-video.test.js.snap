// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withVideo is rendered correctly 1`] = `
<div>
  <video
    height="100%"
    width="100%"
  />
</div>
`;
